app/models/async_transaction/base.rb

```diff
@@ -7,6 +7,8 @@ class Base < ActiveRecord::Base
     REQUESTED = 'requested'
     COMPLETED = 'completed'
 
+    attr_encrypted :metadata, key: Settings.db_encryption_key
```

Comment:
If you add something like

```ruby
before_save :serialize_metadata

def serialize_metadata
  self.metadata = metadata.to_json unless metadata.is_a?(String)
end
```
we can place whatever we'd like inside metadata and it will be automatically serialized to a JSON-encoded string. Then you can take out the `to_json` call in the subclass and it will be handled automatically.

Thoughts?

Reply:
This would also ensure that any other users of this class can't accidentally set this to something other than a string, which the column requires for us to be able to encrypt it.